interpreter/k8s/src/main/scala/io/buoyant/interpreter/k8s/ConfigMapInterpreterInitializer.scala

```diff
@@ -0,0 +1,101 @@
+package io.buoyant.interpreter
+package k8s
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.twitter.finagle.Dtab
+import com.twitter.finagle.Stack.Params
+import com.twitter.finagle.naming.NameInterpreter
+import com.twitter.finagle.param.Label
+import com.twitter.logging.Logger
+import com.twitter.util._
+import io.buoyant.config.types.Port
+import io.buoyant.k8s.ClientConfig
+import io.buoyant.k8s.v1._
+import io.buoyant.namer.{ConfiguredDtabNamer, InterpreterConfig, InterpreterInitializer, Param}
+import java.util.concurrent.atomic.AtomicReference
+
+class ConfigMapInterpreterInitializer extends InterpreterInitializer {
+  val configClass = classOf[ConfigMapInterpreterConfig]
+  override def configId: String = "io.l5d.k8s.configMap"
+}
+
+object ConfigMapInterpreterInitializer extends ConfigMapInterpreterInitializer
+
+case class ConfigMapInterpreterConfig(
+  host: Option[String],
+  port: Option[Port],
+  namespace: Option[String],
+  name: String,
+  filename: String
+) extends InterpreterConfig with ClientConfig {
+
+  @JsonIgnore
+  override def experimentalRequired = true
+
+  @JsonIgnore
+  def portNum = port.map(_.port)
+
+  @JsonIgnore
+  private[this] val log = Logger()
+
+  @JsonIgnore
+  val nsOrDefault = namespace.getOrElse(DefaultNamespace)
+
+  @JsonIgnore
+  val api = {
+    val client = mkClient(Params.empty).configured(Label("configMapInterpreter"))
+    Api(client.newService(dst))
+  }
+
+  @JsonIgnore
+  val nsApi = api.withNamespace(nsOrDefault)
+
+  @JsonIgnore
+  private[this] object Closed extends Throwable
+
+  @JsonIgnore
+  val stateWithClose = Var.async[Activity.State[Dtab]](Activity.Pending) { state =>
+    val closeRef = new AtomicReference[Closable](Closable.nop)
+    val pending = nsApi.configMap(name).get().respond {
+      case Throw(e) => state.update(Activity.Failed(e))
+      case Return(configMap) =>
+        val initState = getDtab(configMap)
+        state.update(Activity.Ok(initState))
+        val (stream, close) = nsApi.configMap(name).watch(None, None, None)
+        closeRef.set(close)
+        val _ = stream.foldLeft(initState) { (dtab, watchEvent) =>
+          val newState: Dtab = watchEvent match {
+            case ConfigMapAdded(a) => getDtab(a)
+            case ConfigMapModified(m) => getDtab(m)
+            case ConfigMapDeleted(_) => Dtab.empty
+            case ConfigMapError(e) =>
+              log.error("k8s watch error: %s", e)
+              dtab
+          }
+          state.update(Activity.Ok(newState))
+          newState
+        }
+    }
+
+    Closable.make { t =>
+      pending.raise(Closed)
+      Closable.ref(closeRef).close(t)
+    }
+  }
```

Comment:
Also, as an aside, one thing we're _not_ doing here that we are doing for IngressCache and K8sDtabStore, is fetching the current resource version before establishing the watch, and then passing that version as the resourceVersion param on the `.watch` call.

If I remember correctly, that's because when I tested this code on Kubernetes, I was running into an infinite loop where the call to watch with a given resource version would fail if the resource version is "too old" (more than ~1m old), which would trigger resetWatches, which would then re-fetch the same resource version, try to re-establish a watch, and run into the same error.

The reason why we don't encounter the loop with IngressCache and K8sDtabStore is that those are watching list endpoints, and I think it's the case that List endpoints always return the system-wide current resource version. For instance:

```
# curl -s localhost:8001/api/v1/namespaces/test/configmaps | jq '.metadata.resourceVersion'
"52871682"
# curl -s localhost:8001/api/v1/namespaces/test/configmaps | jq '.metadata.resourceVersion'
"52871686"
```

Whereas Read endpoints return the resource version for the object is being read:

```
# curl -s localhost:8001/api/v1/namespaces/test/configmaps/namerd-config | jq '.metadata.resourceVersion'
"45625196"
# curl -s localhost:8001/api/v1/namespaces/test/configmaps/namerd-config | jq '.metadata.resourceVersion'
"45625196"
```

So I think it's the correct behavior here to pass in `resourceVersion = None` to the watch call, but I just wanted to call it out because it is weird. Maybe, to be consistent with the IngressCache and K8sDtabStore implementations, we should be calling a different list endpoint altogether (`/api/v1/namespaces`?), and using the resourceVersion from that endpoint to initiate this watch.